### PR TITLE
Add details for Hexadecimal Literals

### PR DESCRIPTION
**Pull request by zoujia-cm**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- The PR is translated from:

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@zoujia-cm Thanks for your contribution! Could you please sign the [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=5619) for this PR to be merged?

File: literal-values.md [hunk 1]
```diff
@@ -169,7 +169,7 @@ X'1z' (z 不是合法的十六进制值)
 0X12AC (0X 必须用小写的 0x)
 ```
 
-对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要有一个数字，可以在前面补一个 0 来避免语法错误。
+对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要有一个数字，如果数字的长度是奇数（比如X'A',X'11A'），可以在前面补一个 0 来避免语法错误。
```
Comment (XuHuaiyu) [inline literal-values.md:172]: 数字的长度 --> val 的长度?

Comment (zoujia-cm) [reply literal-values.md]: > 数字的长度 --> val 的长度?

I have changed "数字的长度" to "val的长度".

**Comment by zoujia-cm** [comment]
@XuHuaiyu please take a look

File: literal-values.md [hunk 2]
```diff
@@ -169,7 +169,7 @@ X'1z' (z 不是合法的十六进制值)
 0X12AC (0X 必须用小写的 0x)
 ```
 
-对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要有一个数字，可以在前面补一个 0 来避免语法错误。
+对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要有一个数字，如果val的长度是奇数（比如X'A',X'11A'），可以在前面补一个 0 来避免语法错误。
```
Comment (XuHuaiyu) [inline literal-values.md:172]: val --> \`val\` ?

Comment (zoujia-cm) [reply literal-values.md]: > val --> `val` ?

changed

**Comment by zoujia-cm** [comment]
@XuHuaiyu  PTAL

File: literal-values.md [hunk 3]
```diff
@@ -169,7 +169,7 @@ X'1z' (z 不是合法的十六进制值)
 0X12AC (0X 必须用小写的 0x)
 ```
 
-对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要有一个数字，可以在前面补一个 0 来避免语法错误。
+对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要是一个数字，如果`val`的长度是奇数（比如X'A',X'11A'），可以在前面补一个 0 来避免语法错误。
```
Comment (yikeke) [inline literal-values.md:172]:
```suggestion
对于使用 `X'val'` 格式的十六进制字面值，`val` 必须要是一个数字，如果 `val` 的长度是奇数（比如 X'A'、X'11A'），可以在前面补一个 0 来避免语法错误。
```

Comment (zoujia-cm) [reply literal-values.md]: changed

Comment (zoujia-cm) [reply literal-values.md]: > 

done

**Pull request merged**